plugin/src/main/java/org/opensearch/ml/sdkclient/RemoteClusterIndicesClient.java

```diff
@@ -0,0 +1,124 @@
+                IndexResponse indexResponse = openSearchClient.index(indexRequest);
+                log.info("Creation status for id {}: {}", indexResponse.id(), indexResponse.result());
+                return new PutDataObjectResponse.Builder().id(indexResponse.id()).created(indexResponse.result() == Created).build();
+            } catch (Exception e) {
+                throw new OpenSearchException("Error occurred while indexing data object", e);
+            }
+        }), executor);
+    }
+
+    @Override
+    public CompletionStage<GetDataObjectResponse> getDataObjectAsync(GetDataObjectRequest request, Executor executor) {
+        return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<GetDataObjectResponse>) () -> {
+            try {
+                GetRequest getRequest = new GetRequest.Builder().index(request.index()).id(request.id()).build();
+                log.info("Getting {} from {}", request.id(), request.index());
+                @SuppressWarnings("rawtypes")
+                GetResponse<Map> getResponse = openSearchClient.get(getRequest, Map.class);
+                if (!getResponse.found()) {
+                    return new GetDataObjectResponse.Builder().id(getResponse.id()).build();
+                }
+                String json = new ObjectMapper().setSerializationInclusion(Include.NON_NULL).writeValueAsString(getResponse.source());
+                log.info("Retrieved data object");
+                XContentParser parser = JsonXContent.jsonXContent
+                    .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, json);
+                return new GetDataObjectResponse.Builder().id(getResponse.id()).parser(Optional.of(parser)).build();
+            } catch (Exception e) {
+                throw new OpenSearchException(e);
+            }
+        }), executor);
+    }
+
+    @Override
+    public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(DeleteDataObjectRequest request, Executor executor) {
+        return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<DeleteDataObjectResponse>) () -> {
+            try {
+                DeleteRequest deleteRequest = new DeleteRequest.Builder().index(request.index()).id(request.id()).build();
+                log.info("Deleting {} from {}", request.id(), request.index());
+                DeleteResponse deleteResponse = openSearchClient.delete(deleteRequest);
+                log.info("Deletion status for id {}: {}", deleteResponse.id(), deleteResponse.result());
+                ShardInfo shardInfo = new ShardInfo(
+                    deleteResponse.shards().total().intValue(),
+                    deleteResponse.shards().successful().intValue()
+                );
+                return new DeleteDataObjectResponse.Builder()
+                    .id(deleteResponse.id())
+                    .shardId(deleteResponse.index())
```

Comment:
[nit] if we are just trying to put some random value for other remote stores, let's a comment.

Reply:
There is a comment in the `DeleteDataObjectResponse` javadocs.  Where else should this be commented?